### DISCUSSION TITLE
filenodes->add && PDOKAN_IO_SECURITY_CONTEXT

### DISCUSSION THREAD

**Discussion by Liuyadi-Code** [dokan-dev/dokany#1215] [Q&A]
\I'd like to ask how I need to set the parameters for adding files using filenodes->add(), specifically the PDOKAN_IO_SECURITY_CONTEXT parameter. Can you provide an example?

**Comment by Liryna** [comment]
This is about memfs sample correct ? The param is optional https://github.com/dokan-dev/dokany/blob/master/samples/dokan_memfs/filenode.cpp#L40-L43 but if you want to learn how to manipulate your own (the memfs only use the one given by the system during create)
See here what the security context is [here](https://github.com/dokan-dev/dokany/blob/69e3d88c693da08fc1e9c540c162877880eac155/dokan/dokan.h#L272).
